### PR TITLE
Make typesafe

### PR DESCRIPTION
**Pull request by juliarei**

### CODE REVIEW

File: lib/controllers/carplay_controller.dart [hunk 1]
```diff
@@ -50,16 +50,9 @@ class FlutterCarplayController {
...
-            //   for (final (tabIndex, tab) in tabBarTemplate.templates.indexed) {
-            //     if (tab.uniqueId == elementId) {
-            //       tabBarTemplate.templates[tabIndex] = updatedTemplate;
-            //       return;
-            //     }
-            //   }
             case final CPInformationTemplate informationTemplate:
               if (informationTemplate.uniqueId == elementId) {
-                template = updatedTemplate;
+                template = updatedTemplate as CPTemplate;
```
Comment (mikeroneer) [inline lib/controllers/carplay_controller.dart]: `updatedTemplate` is of type `CPInfomationTemplate` which derives from `CPTemplate`, so a type conversation should not be needed

File: lib/controllers/carplay_controller.dart [hunk 2]
```diff
@@ -90,7 +83,7 @@ class FlutterCarplayController {
               }
             case final CPListTemplate listTemplate:
               if (listTemplate.uniqueId == elementId) {
-                template = updatedTemplate;
+                template = updatedTemplate as CPTemplate;
```
Comment (mikeroneer) [inline lib/controllers/carplay_controller.dart]: same as above

File: lib/controllers/carplay_controller.dart (hunk 1 above)
Comment (juliarei) [reply lib/controllers/carplay_controller.dart]: Right, removed it.

File: lib/controllers/carplay_controller.dart (hunk 2 above)
Comment (juliarei) [reply lib/controllers/carplay_controller.dart]: done

**Pull request merged**